qiskit/converters/ast_to_dag.py

```diff
@@ -67,6 +67,20 @@ def ast_to_dag(ast):
 
     Raises:
         QiskitError: if the AST is malformed.
+
+    Example:
+        .. code-block::
+
+            from qiskit.converters import ast_to_dag
+            from qiskit import qasm
+            from qiskit.visualization import dag_drawer
+            %matplotlib inline
+
+            # local file with QASM code
+            ast = qasm.Qasm(filename=self._get_resource_path('example.qasm',
+                                                        Path.QASMS)).parse()
```

Comment:
What is `self` here? One thing we could do here is do is just use `QuantumCircuit.qasm()` to generate a qasm string and pass that to the parser. For example, make the example:

```
q = QuantumRegister(3, 'q')
c = ClassicalRegister(3, 'c')
circ = QuantumCircuit(q, c)
circ.h(q[0])
circ.cx(q[0], q[1])
circ.measure(q[0], c[0])
circ.rz(0.5, q[1]).c_if(c, 2)
qasm_str = circ.qasm()
ast = qasm.Qasm(data=qasm_str).parse()
dag = ast_to_dag(ast)
dag_drawer(dag)
```
and that would also let us use `jupyter-execute` here too.

Reply:
Got it - I had tried using a QASM string directly with `jupyter-execute`, but Sphinx didn't seem to like the syntax. In other examples on Github, the function was called with a separate file so I just left it as that. I've updated the PR with this change.